lib/game_team.rb

```diff
@@ -133,53 +133,33 @@ def self.fewest_tackles(season_id)
     fewest_tackles.first
   end
 
-  def self.best_offense
-    #grouped by team_id with keys being the team_id and values is an array of games
+  def self.total_goals_per_team
     grouped_team = all.group_by{|game| game.team_id}
-    #loop through the values (games) and set them equal to the average of goals
-    team_averaged_goals = grouped_team.map do |ids, games|
-      goals_per_game = games.map {|game| game.goals}
-      games = (goals_per_game.sum / goals_per_game.length.to_f)
+    grouped_team.keys.each_with_object({}) do |team_id , hash|
+      hash[team_id] = (grouped_team[team_id].sum(&:goals) / grouped_team[team_id].length.to_f).round(2)
```

Comment:
each_with_object creates a hash and iterates through each item of grouped_team therefore giving me access to the hash without calling `Hash.new`